pyiron/base/project/generic.py

```diff
@@ -121,7 +124,14 @@ class Project(ProjectPath):
 
     """
 
-    def __init__(self, path="", user=None, sql_query=None):
+    def __init__(self, path="", user=None, sql_query=None, default_working_directory=False):
+        if default_working_directory:
+            inputdict = Notebook.get_custom_dict()
+            if "project_dir" in inputdict.keys():
+                path = inputdict["project_dir"]
+            else:
+                path = "."
```

Comment:
I guess we can give a warning here, because I agree with you that this is not recommended.

Reply:
(I mean `Project('.')` is not recommended)

Reply:
Maybe just to clarify: I guess if the user explicitly writes `Project(path='.')`, it's fine, because then it's likely that they know what they're doing, but if `path='.'` is automatically set because `default_working_directory` was turned on, then there's the danger that the user doesn't know what's happening.

Reply:
If no script job is found it at least prints that no HDF5 file is found: 
https://github.com/pyiron/pyiron/blob/script_job/pyiron/base/job/external.py#L53

Reply:
Okay true. Maybe that's good enough. But shouldn't it be a warning anyway?

Reply:
I updated it to use a warning now.